src/openforms/js/components/admin/form_design/MissingComponentTranslationsWarning.js

```diff
@@ -0,0 +1,141 @@
+import {ChangelistColumn, ChangelistTable} from 'components/admin/tables';
+import {
+  extractComponentLiterals,
+  getSupportedLanguages,
+} from 'components/formio_builder/translation';
+
+const LANGUAGES = getSupportedLanguages();
+
+const extractTranslatableValues = configuration => {
+  let translatableValues = [];
+  FormioUtils.eachComponent(
+    configuration.components,
+    component => {
+      const literals = extractComponentLiterals(component);
+      literals.forEach(literal => {
+        translatableValues.push({
+          componentKey: component.key,
+          componentLabel: component.label,
+          literal: literal,
+        });
+      });
+    },
+    true
+  );
+  return translatableValues;
+};
+
+const MissingComponentTranslationsTable = ({children: missingTranslations}) => (
+  <ChangelistTable data={missingTranslations}>
+    <ChangelistColumn objProp="componentKey">
+      <FormattedMessage description="Key of the Form component" defaultMessage="Component key" />
+    </ChangelistColumn>
+
+    <ChangelistColumn objProp="componentLabel">
+      <FormattedMessage
+        description="Label of the Form component"
+        defaultMessage="Component label"
+      />
+    </ChangelistColumn>
+
+    <ChangelistColumn objProp="language">
+      <FormattedMessage description="Readable label for the language" defaultMessage="Language" />
+    </ChangelistColumn>
+
+    <ChangelistColumn objProp="literal">
+      <FormattedMessage description="Literal to be translated" defaultMessage="Literal" />
+    </ChangelistColumn>
+  </ChangelistTable>
+);
+
+MissingComponentTranslationsTable.propTypes = {
+  children: PropTypes.arrayOf(
+    PropTypes.shape({
+      componentKey: PropTypes.string.isRequired,
+      componentLabel: PropTypes.string.isRequired,
+      language: PropTypes.string.isRequired,
+      literal: PropTypes.string.isRequired,
+    })
+  ),
+};
+
+const MissingComponentTranslationsWarning = ({configuration, componentTranslations}) => {
+  const languageCodeMapping = Object.fromEntries(LANGUAGES);
+
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const onShowModal = event => {
+    event.preventDefault();
+    setModalOpen(true);
+  };
+
+  const translatableValues = extractTranslatableValues(configuration);
+
+  let missingTranslations = [];
+  for (const entry of translatableValues) {
+    for (const [languageCode, _languageLabel] of LANGUAGES) {
+      let translations = componentTranslations[languageCode] || {};
+      if (!translations[entry.literal])
+        missingTranslations.push({language: languageCodeMapping[languageCode], ...entry});
+    }
+  }
+  missingTranslations.sort((a, b) => {
+    if (a.componentKey !== b.componentKey) {
+      return a.componentKey.localeCompare(b.componentKey);
+    }
+    if (a.language !== b.language) {
+      return a.language.localeCompare(b.language);
+    }
+    return a.literal.localeCompare(b.literal);
+  });
```

Comment:
Added some more sorting to group/sort by component, then language, then literal. This makes sense as in the UI you open the component, then navigate to a particular language tab and fix all the relevant literals per language.